llvm/include/llvm/ADT/FunctionExtras.h

```diff
@@ -319,6 +319,9 @@ template <typename ReturnT, typename... ParamTs> class UniqueFunctionBase {
 
 #ifndef NDEBUG
     // In debug builds, we also scribble across the rest of the storage.
+#ifndef _LIBCPP_HAS_NO_ASAN
+    __asan_unpoison_memory_region(RHS.getInlineStorage(), InlineStorageSize);
+#endif
     memset(RHS.getInlineStorage(), 0xAD, InlineStorageSize);
```

Comment:
I'd rather not scribble at all with asan.
And probably use non libc++ specific way do detect ASAN. Make it consistent with the surrounding code.

Reply:
What is a general way of checking if compiling with ASan? `defined(ADDRESS_SANITIZER) || defined(__SANITIZE_ADDRESS__)`?

Reply:
I used those, if there is something better, please let me know in any new comment.